twitter/twitter_client.bal

```diff
@@ -0,0 +1,1057 @@
+    public remote function search(string query, int? count = (), string? lang = (), boolean? includeEntities = (),
+                                  int? sinceId = (), int? maxId = (), string? locale = (), string? geocode = (),
+                                  string? until = (), SearchResultType? resultType = ()) returns Search|error;
+
+    # Returns a collection of the most recent Tweets and Retweets posted by the authenticating user and
+    # the users they follow.
+    #
+    # + count - Specifies the number of records to retrieve.
+    # + sinceId - Returns results with an ID greater than (that is, more recent than) the specified ID.
+    # + maxId - Returns results with an ID less than (that is, older than) or equal to the specified ID.
+    # + trimUser - When set to true, each Tweet returned in a timeline will include a user object including only the
+    #              status authors numerical ID. Omit this parameter to receive the complete user object.
+    # + excludeReplies - This parameter will prevent replies from appearing in the returned timeline.
+    # + includeEntities - The entities node will not be included when set to false.
+    # + return - Returns a collection of Tweet object when successful, else returns an error.
+    public remote function getHomeTimelineTweets(int? count = (), int? sinceId = (), int? maxId = (),
+                                                 boolean? trimUser = (), boolean? excludeReplies = (),
+                                                 boolean? includeEntities = ()) returns Tweet[]|error;
+
+    # Returns the 20 most recent mentions (Tweets containing a users's @screen_name) for the authenticating user.
+    # The timeline returned is the equivalent of the one seen when you view your mentions on twitter.com.
+    #
+    # + count - Specifies the number of Tweets to try and retrieve, up to a maximum of 200.
+    # + sinceId - Returns results with an ID greater than (that is, more recent than) the specified ID.
+    # + maxId - Returns results with an ID less than (that is, older than) or equal to the specified ID.
+    # + trimUser - When set to true, each tweet returned in a timeline will include a user object including only
+    #              the status authors numerical ID. Omit this parameter to receive the complete user object.
+    # + includeEntities - The entities node will not be included when set to false.
+    # + return - Returns a collection of Tweet object when successful, else returns an error.
+    public remote function getTweetMentions(int? count = (), int? sinceId = (), int? maxId = (), boolean? trimUser = (),
+                                            boolean? includeEntities = ()) returns Tweet[]|error;
+};
+
+# Twitter User Client object.
+#
+# + httpClient - HTTP Client.
+# + twitterClient - Twitter Client.
+public type TwitterUserClient client object {
+    http:Client httpClient;
+    TwitterClient|error twitterClient;
```

Comment:
Same as previous comment regarding setting the error returned on new-ing up instead of returning it.